components/ChipTextField/examples/MDCChipTextField.m

```diff
@@ -29,144 +37,254 @@ @implementation MDCChipTextField
-    self.leadingConstraint = [NSLayoutConstraint constraintWithItem:self.chipsView
-                                                          attribute:NSLayoutAttributeLeading
-                                                          relatedBy:NSLayoutRelationEqual
-                                                             toItem:chip
-                                                          attribute:NSLayoutAttributeLeading
-                                                         multiplier:1.0
-                                                           constant:0];
-    [self.chipsView addConstraint:self.leadingConstraint];
-  } else {
-    [self.chipsView addConstraint:[NSLayoutConstraint constraintWithItem:lastChip
-                                                               attribute:NSLayoutAttributeTrailing
-                                                               relatedBy:NSLayoutRelationEqual
-                                                                  toItem:chip
-                                                               attribute:NSLayoutAttributeLeading
-                                                              multiplier:1.0
-                                                                constant:0]];
-  }
+  MDCChipView *chipView = [[MDCChipView alloc] init];
+  chipView.titleLabel.text = text;
+  chipView.translatesAutoresizingMaskIntoConstraints = NO;
+
+  [self.chipsContainerView appendChipView:chipView];
 
   // recalculate the layout to get a correct chip frame values
-  [self.chipsView layoutIfNeeded];
-  self.insetX = CGRectGetMaxX(chip.frame);
-  [self.chips addObject:chip];
+  [self.chipsContainerView layoutIfNeeded];
+  [self.chipViews addObject:chipView];
+}
 
-  self.leftViewMode = UITextFieldViewModeAlways;
+- (void)clearChipsContainerOffset {
+  [self layoutIfNeeded];
+  CGFloat differenceBetweenContentSizeAndFrameSizeOnXAxis = self.chipsContainerView.contentSize.width - CGRectGetWidth(self.chipsContainerView.bounds);
+  self.chipsContainerHorizontalWidthForTextEditing = differenceBetweenContentSizeAndFrameSizeOnXAxis;
+  self.chipsContainerHorizontalOffsetForTextEditing = differenceBetweenContentSizeAndFrameSizeOnXAxis;
 }
 
+#pragma mark - Text Editing Handler
+
 - (void)chipTextFieldTextDidChange {
+  // Set contentOffset
   [self deselectAllChips];
   [self setupEditingRect];
 }
 
 - (void)setupEditingRect {
-  CGRect textRect = [self textRectForBounds:self.bounds];
+  CGRect textRect = [self editingRectForBounds:self.bounds];
   UITextRange *textRange = [self textRangeFromPosition:self.beginningOfDocument
                                             toPosition:self.endOfDocument];
   CGRect inputRect = [self firstRectForRange:textRange];
+  // HACK: use a constant here
+  CGFloat textHorizontalDelta = textRect.size.width - inputRect.size.width - 17;
+  CGFloat offsetAdjustment = textHorizontalDelta;
+//  // HACK: this assumption is also bad
+//  UIScrollView *fieldEditor = nil;
```

Comment:
Should we remove all of this?

Reply:
Addressed in 233e4691a7c8f02864dbca983d6d6203b87caf50 to make the comment clear. We probably need to follow up and solve the problem here. Leaving the code here would be good for reference for now since it is in Alpha.